tests/unittests/test_lookback_window.py

```diff
@@ -0,0 +1,150 @@
+        Context.config = config
+        Context.new_counts['balance_transactions'] = 1
+        sync_stream("balance_transactions")
+        # expected start_date should be the bookmark time - `lookback`(default lookback)
+        expected_start_window = bookmark_time - IMMUTABLE_STREAM_LOOKBACK
+        mock_epoch_to_dt.assert_called_with(expected_start_window)
+
+    def test_default_value_lookback_events(self, mock_get_bookmark_for_stream, mock_sync_substream, mock_dt_to_epoch, mock_epoch_to_dt, mock_get, mock_metadata_map, mock_get_catalog_entry, mock_paginate, mock_convert_dict_to_stripe_object, mock_reduce_foreign_keys):
+        '''Verify that the lookback window is 600 by default if nothing is provided in the config.'''
+        config = {"client_secret": "test_secret", "account_id": "test_account", "start_date": "2022-02-17T00:00:00"}
+        Context.config = config
+        Context.new_counts['events'] = 1
+        sync_stream("events")
+        # expected start_date should be the bookmark time - `lookback`(default lookback)
+        expected_start_window = bookmark_time - IMMUTABLE_STREAM_LOOKBACK
+        mock_epoch_to_dt.assert_called_with(expected_start_window)
+
+    def test_config_provided_value_lookback_events(self, mock_get_bookmark_for_stream, mock_sync_substream, mock_dt_to_epoch, mock_epoch_to_dt, mock_get, mock_metadata_map, mock_get_catalog_entry, mock_paginate, mock_convert_dict_to_stripe_object, mock_reduce_foreign_keys):
+        '''Verify that the lookback window is correctly passed when mentioned in the config.'''
+        config = {"client_secret": "test_secret", "account_id": "test_account", "start_date": "2022-02-17T00:00:00", "lookback_window": 300}
+        Context.config = config
+        Context.new_counts['events'] = 1
+        sync_stream("events")
+        # expected start_date should be the bookmark time - `lookback`(lookback passed in the config)
+        expected_start_window = bookmark_time - config.get('lookback_window')
+        mock_epoch_to_dt.assert_called_with(expected_start_window)
+        
+    def test_empty_string_in_config_lookback_events(self, mock_get_bookmark_for_stream, mock_sync_substream, mock_dt_to_epoch, mock_epoch_to_dt, mock_get, mock_metadata_map, mock_get_catalog_entry, mock_paginate, mock_convert_dict_to_stripe_object, mock_reduce_foreign_keys):
+        '''Verify that the lookback window is correctly passed when empty string is passed in the config.'''
+        config = {"client_secret": "test_secret", "account_id": "test_account", "start_date": "2022-02-17T00:00:00", "lookback_window": ''}
+        Context.config = config
+        Context.new_counts['events'] = 1
+        sync_stream("events")
+        # expected start_date should be the bookmark time - `lookback`(default lookback)
+        expected_start_window = bookmark_time - IMMUTABLE_STREAM_LOOKBACK
+        mock_epoch_to_dt.assert_called_with(expected_start_window)
```

Comment:
Empty string '' is an invalid value so this test should fail. Here you may need to change the assertion accordingly.

Reply:
As per the discussion with @KrisPersonal, the current implementation makes sense. Hence moving forward with the same